fast64_internal/oot/data/oot_object_data.py

```diff
@@ -0,0 +1,91 @@
+from .oot_getters import getRoot, getEnumList
+from os import path
+
+# Note: "object" in this context refers to an OoT Object file (like ``gameplay_keep``)
+
+
+class OoT_ObjectData:
+    """Everything related to OoT objects"""
+
+    from dataclasses import dataclass
+
+    @dataclass
+    class OoT_Object:
+        id: str
+        key: str
+        name: str
+
+    def __init__(self):
+        # Path to the ``ObjectList.xml`` file
+        self.objectXML: str
+
+        # general object list
+        self.objectList: list[self.OoT_Object] = []
+
+        # list of tuples used by Blender's enum properties
+        self.ootEnumObjectID: list[tuple] = []
+        self.ootEnumObjectIDLegacy: list[tuple] = []  # for old blends
+
+        self.initObjectLists()
+
+    def initObjectLists(self):
+        """Reads the XML and make a list of the useful data to keep"""
+        self.objectXML = path.dirname(path.abspath(__file__)) + "/xml/ObjectList.xml"
+        for obj in getRoot(self.objectXML).iterfind("Object"):
+            self.objectList.append(self.OoT_Object(obj.attrib["ID"], obj.attrib["Key"], obj.attrib["Name"]))
+        self.ootEnumObjectID, self.ootEnumObjectIDLegacy = getEnumList(self.objectList, "Custom Object")
+
+    def upgradeObjectInit(self, obj, objectList):
+        """Object upgrade logic"""
+        if obj.data is None and obj.ootEmptyType == "Room":
+            for i in range(len(obj.objectList)):
+                obj.objectList[i].objectIDLegacy = obj.objectList[i].objectID
+                for object in objectList:
+                    if obj.objectList[i].objectIDLegacy == object.id:
+                        obj.objectList[i].objectID = object.key
+
+            obj.fast64.oot.version = obj.fast64.oot.cur_version
```

Comment:
setting the version probably should be done somewhere else, as this function deals specifically with version 0 -> version 1 and for the object list

probably in `upgrade_changed_props` ?
(`SM64_ObjectProperties.upgrade_changed_props` and another one does so it looks like)